overlord/devicestate/devicestate.go

```diff
@@ -320,16 +320,31 @@ func extractDownloadInstallEdgesFromTs(ts *state.TaskSet) (firstDl, lastDl, firs
 
 func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Model, deviceCtx snapstate.DeviceContext, fromChange string) ([]*state.TaskSet, error) {
 	userID := 0
+	var tss []*state.TaskSet
 
 	// adjust kernel track
-	var tss []*state.TaskSet
-	if current.KernelTrack() != new.KernelTrack() {
+	if current.Kernel() == new.Kernel() && current.KernelTrack() != new.KernelTrack() {
```

Comment:
these will need changes soon  for the UC20 work, but it will done together when I remove/change some of these accessors, in favor of KernelSnap